core/src/main/java/org/infinispan/util/concurrent/NonBlockingManager.java

```diff
@@ -13,19 +14,44 @@ public interface NonBlockingManager {
     * Schedules the supplier that is executed after the <b>initialDelay</b> period and subsequently runs <b>delay</b>
     * after the previous stage completes. The supplier must not block the thread in which it runs and should immediately
     * return to avoid blocking the scheduling thread.
+    * <p>
+    * This method will not reschedule the task if the supplier or the stage throw any Throwable.
+    *
     * @param supplier non-blocking operation supplier.
     * @param initialDelay period of time before the supplier is invoked.
     * @param delay delay between subsequent supplier invocations.
     * @param unit time unit for delays.
     * @return an AutoCloseable that cancels the scheduled task.
     */
-   AutoCloseable scheduleWithFixedDelay(Supplier<CompletionStage<?>> supplier, long initialDelay, long delay, TimeUnit unit);
+   default AutoCloseable scheduleWithFixedDelay(Supplier<CompletionStage<?>> supplier, long initialDelay, long delay, TimeUnit unit) {
+      return scheduleWithFixedDelay(supplier, initialDelay, delay, unit, null);
+   }
+
+   /**
+    * Schedules the supplier that is executed after the <b>initialDelay</b> period and subsequently runs <b>delay</b>
+    * after the previous stage completes. The supplier must not block the thread in which it runs and should immediately
+    * return to avoid blocking the scheduling thread.
+    * <p>
+    * This method will not be rescheduled if the supplier throws any Throwable, note this is not the stage.
```

Comment:
Hmm this isn't clear to me. How about something like: "By default the supplier won't be rescheduled if the supplier completes exceptionally. Optionally, ..."

Reply:
Sure

Reply:
Actually "By default" doesn't apply here. It is only rescheduled if the Stage completes exceptionally. If the supplier itself throws an exception it terminates. I will try to make it more clear. Unfortunately, I was trying to be a bit more vague in some  cases.